lib/src/main/java/com/dynatrace/metric/util/MetricValues.java

```diff
@@ -143,16 +143,16 @@ public String serialize() {
     }
   }
 
-  private static final Pattern regexTrailingZeroesOrDots = Pattern.compile("[0.]+$");
+  private static final Pattern regexTrailingZeroes = Pattern.compile("[0]+$");
 
   static String formatDouble(double d) {
-    String formatted = String.format("%.6f", d);
-    // trim trailing zeros and dots
-    formatted = regexTrailingZeroesOrDots.matcher(formatted).replaceAll("");
-    if (formatted.isEmpty()) {
-      // everything was trimmed away, number was 0.0000
-      return "0";
+    String formatted = String.valueOf(d);
+    // trim trailing zeros
+    formatted = regexTrailingZeroes.matcher(formatted).replaceAll("");
```

Comment:
This will fail for `"1.0E10"`, for example - please add a test for that. Also I'm not sure if it would ever print a number with neither `.` nor `E` for a certain length of the integer part with no fractional part.
I suggest to use `"\.0+$"`.